File: find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
// class Solution {
// public:
//     int findMin(vector<int>& nums) {\
//         int ans=nums[0];
//         int low=0,high=nums.size()-1;
//         int n=nums.size();
//         while(low<high)
//         {
//             int mid=low+(high-low)/2;
//             if((nums[mid]<nums[(mid+1)%n])&&(nums[mid]<nums[(mid-1+n)%n]))
//             return nums[mid];
//             else if(nums[mid]>=nums[low])
//                low=mid+1;
//             else if(nums[mid]<=nums[high])
//                high=mid-1;
               
//         }
//                return ans;
        
//     }
// };


class Solution {
public:
    int findMin(vector<int>& nums) {
        int low=0;
        int high=nums.size()-1;
        
        while(low<high){
            int mid=low+(high-low)/2;
            
            if(nums[mid]<nums[high]){
                high=mid;
            }
            else{
                low=mid+1;
            }
        }
       
        return nums[high];
    }
};

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        lo = 0
        hi = len(nums) - 1
        
        while lo <= hi:
            mid = lo + (hi - lo) // 2
            
            # If we find the minimum
            if mid > 0 and nums[mid] < nums[mid - 1]:
                return nums[mid]
            
            # Check which half to search
            if nums[mid] > nums[hi]:
                lo = mid + 1
            else:
                hi = mid - 1
        
        # When the loop ends, lo should point to the minimum element
        return nums[lo]
